### PR TITLE
Commit transactions before sending response

### PR DESCRIPTION
**Pull request by seallard**
## Description
The transaction management approach taken in https://github.com/Clinical-Genomics/trailblazer/pull/299 is not ideal since the transactions are committed only after responses are sent. This PR restores the original commits in the CRUD handlers.

By committing before we return a response, we ensure that the caller receives an error if something goes wrong against the database.

### Fixed
- Commit transactions before sending response.

### How to test
- [ ] Ensure cigrid views work and data is written
- [ ] Ensure cli works and data is written

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by diitaz93 — APPROVED** [review]
LGTM 👍

File: trailblazer/store/crud/delete.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def delete_analysis_jobs(self, analysis: Analysis) -> None:
         session: Session = get_session()
         for job in analysis.jobs:
             session.delete(job)
+        session.commit()
```
Comment (Vince-janv) [inline trailblazer/store/crud/delete.py:23]: Do we want this here, or should we perform the commit in the CLI/Endpoint level? If possible I would prefer to have it on the same level for all

**Review by Vince-janv — APPROVED** [review]
Nice 👍

File: trailblazer/store/crud/delete.py (hunk 1 above)
Comment (seallard) [reply trailblazer/store/crud/delete.py:23]: Me too, but I don't like cluttering the cli and api with these commits and I don't want to spend more time thinking about it :sweat_smile: I'm leaving it for now as is, if someone (probably me in the future) feel like they have time and energy to refactor it

Comment (seallard) [reply trailblazer/store/crud/delete.py:23]: The session/transaction management should not be done in the controller or CLI layer either. I think one way to go is to use the `teardown_request` decorator https://flask.palletsprojects.com/en/3.0.x/api/#flask.Flask.teardown_request. But I'm pretty over dealing with this. Feel free to pick it up if you are motivated :muscle:

**Comment by seallard** [comment]
Views work, fields can be written. CLI works, scan and write.

**Pull request merged**